Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -542,6 +542,39 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return handleEffectSpecifiers(node)
   }
 
+  public override func visit(_ node: GenericRequirementSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let unexpected = node.unexpectedBetweenBodyAndTrailingComma,
+      let token = unexpected.first?.as(TokenSyntax.self),
+      token.tokenKind == .binaryOperator("&&"),
+      let trailingComma = node.trailingComma,
+      trailingComma.hasError,
+      let previous = node.unexpectedBetweenBodyAndTrailingComma?.previousToken(viewMode: .sourceAccurate)
+    {
+
+      addDiagnostic(
+        unexpected,
+        .expectedCommaInWhereClause,
+        fixIts: [
+          FixIt(
+            message: ReplaceTokensFixIt(replaceTokens: [TokenSyntax(token)!], replacement: .commaToken()),
```

Comment:
Why do you need to do `TokenSyntax(token)` here? Isn’t `token` already of type `TokenSyntax`?